src/main/java/hudson/plugins/audit_trail/LogFileDailyRotationAuditLogger.java

```diff
@@ -0,0 +1,180 @@
+                    initInstant = Instant.parse(matcher.group(0) + "T00:00:00.00Z").atZone(ZoneId.systemDefault());
+                }
+            }
+        }
+        // Initialize initInstant to the current instant
+        if (initInstant == null) {
+            initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        }
+        configure();
+    }
+
+    String computePattern() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd").withZone(ZoneId.systemDefault());
+        String formattedInstant = formatter.format(initInstant);
+        String computedFileName = String.format("%s-%s", FilenameUtils.getName(basePattern.toString()) , formattedInstant);
+        Path parentFolder = basePattern.getParent();
+        if (parentFolder != null) {
+            return parentFolder.resolve(computedFileName).toString();
+        }
+        return computedFileName;
+    }
+
+    private boolean shouldRotate() {
+        return ZonedDateTime.now().isAfter(initInstant.plus(Duration.ofDays(1)));
+    }
+
+    /**
+     * Rotates the daily rotation logger
+     */
+    private void rotate() {
+        if (getHandler() != null) {
+            getHandler().close();
+        }
+        initInstant = ZonedDateTime.now().truncatedTo(ChronoUnit.DAYS);
+        configure();
+        // After rotating remove old files
+        removeOldFiles();
+    }
+
+    private void removeOldFiles() {
+        Path directoryPath = basePattern.getParent();
+        if (directoryPath != null) {
+            Collection<File> files = FileUtils.listFiles(directoryPath.toFile(), new RegexFileFilter(".*" + FilenameUtils.getName(basePattern.toString()) + DAILY_ROTATING_FILE_REGEX_PATTERN), DirectoryFileFilter.DIRECTORY);
+            if (files.size() > getCount()) {
+                List<File> orderedList = files.stream().sorted(LASTMODIFIED_REVERSE).collect(Collectors.toList());
+                List<File> toDelete = orderedList.subList(getCount(), orderedList.size());
+                for (File file : toDelete) {
+                    if (!file.delete()) {
+                        LOGGER.log(Level.SEVERE, "File {0} could not be removed on rotate overation", file.getName());
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void log(String event) {
+        // to avoid synchronizing the whole method
+        if (shouldRotate()) {
+            synchronized (this) {
+                if (shouldRotate()) rotate();
+            }
+        }
+        super.log(event);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        LogFileDailyRotationAuditLogger that = (LogFileDailyRotationAuditLogger) o;
+
+        if (initInstant != null ? !initInstant.equals(that.initInstant) : that.initInstant != null) return false;
```

Comment:
The `equals` doesn't take into account the fields of the super class.
For instance that means that the plugin would consider two different loggers configured with a different `separator` as equals (since the `separator` field is configured in the super). I'm not sure it would cause a bug currently since IIRC we only push the loggers in a list, but it could be cause for subtle bugs in the future (if we add a feature needing to compare loggers, or we store them in a `Set` for instance).

I'd suggest:
* in this class: call the super (same in `LogFileAuditLogger`), and same for the hashcode.
* in the super class generate equals and hashcode (let your IDE do it).